### PR TITLE
AMLII-1359: Adds a local docker-compose env to test JMXFetch inside the agent

### PR DESCRIPTION
**Pull request by scottopell**
This adds a `docker-compose` based environment where local changes to JMXFetch can be easily tested.

No particular setup required, simply running `docker compose up` will build and run the necessary containers.

A future improvement would be to allow custom configurations for `misbehaving-jmx-server` so that more than 1 bean can be tested, however that is left for future improvement.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    labels:
+      com.datadoghq.ad.checks: '{"misbehaving":{"init_config":{"is_jmx":true},"instances":[{"host":"jmx-test-server","port":"1099","collect_default_jvm_metrics":false,"max_returned_metrics":300000,"conf":[{"include":{"domain":"Bohnanza"}}]}]}}'
+  datadog:
+    build:
+      context: .
+      dockerfile: ./Dockerfile.agentbase
+    pid: host
+    environment:
+     - DD_API_KEY=000000001
+     - DD_JMX_TELEMETRY_ENABLED=true
```
Comment (carlosroman) [inline docker-compose.yaml:24]:
```suggestion
    env_file: .env # has entry with DD_API_KEY=000000001
    environment:
     - DD_JMX_TELEMETRY_ENABLED=true
```

**Review by carlosroman — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -72,6 +74,15 @@ the recommended JDK version for development, use `sdk env` to activate it.
 If you set the system property `-Djmxfetch.filelinelogging=true`, this will enable all log output to
 include the line number which emitted a given log.
 
+### Local Testing
+You can utilize the provided testing server `misbehaving-jmx-server` with the
+preset `./docker-compose.yaml` file. This will run two containers, one is the
```
Comment (apigirl) [inline README.md:79]:
```suggestion
preset `./docker-compose.yaml` file. This runs two containers, one is the
```

**Review by apigirl — COMMENTED** [review]
tiny editorial nit

**Comment by scottopell** [comment]
@apigirl if your concern is addressed, can I get an approval here please?

**Review by aliciascott — APPROVED** [review]

**Pull request merged**